### DISCUSSION TITLE
Langium-Sprotty exemples on the git?

### DISCUSSION THREAD

**Discussion by Olivier-D09** [eclipse-langium/langium#1409] [Q&A] [answered]
hi, I'm looking for examples of Langium representation using sprotty, I've searched on the git and haven't found any, does anyone know where I can find some, if any on github?

**Comment by spoenemann** [comment, marked as answer]
The sprotty-vscode integration has such an example: https://github.com/eclipse-sprotty/sprotty-vscode/tree/master/examples

**Comment by jdeantoni** [comment]
Hi, thanks for the link. Having the same question I had a look at the sprotty example and I saw that the sprotty code is "rebundling" the LSP. 

Is there a way to add a sprotty representation to a langium language without modifying/rebundling the orginal LSP ?

**Reply by spoenemann** [reply]
No, because the LSP doesn't have any representation of AST contents. You need some way to transform the AST into a diagram model. The easiest way is to do it within the language server, but an alternative could be to transfer the AST to the client using JsonSerializer.

**Reply by jdeantoni** [reply]
I see. I'll give a try to both !
Thanks for the answer !

**Reply by spoenemann** [reply]
Here's an example of how that can work:

https://github.com/eclipse-langium/langium/blob/f6889f23b6462dd388dfc9f1e787a44bd29a7c1d/examples/statemachine/src/language-server/main-browser.ts#L25-L38

**Reply by jdeantoni** [reply]
Thanks ! That surely helps !
